nautobot/core/templatetags/helpers.py

```diff
@@ -725,6 +725,27 @@ def render_address(address):
     return HTML_NONE
 
 
+@library.filter()
+@register.filter()
+def render_button_class(value):
+    """
+    Render a string as a styled HTML button using Bootstrap classes.
+
+    Args:
+        value (str): A string representing the button class (e.g., 'primary').
+
+    Returns:
+        str: HTML string for a button with the given class.
+
+    Example:
+        >>> render_button_class("primary")
+        '<button class="btn btn-primary">primary</button>'
+    """
+    if value:
+        return format_html('<button class="btn btn-{}">{}</button>', value.lower(), value.lower())
```

Comment:
The original template is
```
<td><button class="btn btn-{{ object.button_class }}">{{ object.button_class | title }}</button></td>
```
So I think we need to apply `title` filter to the second `value` instead of `value.lower()`, you can see the difference in the screenshots you provided as well.

Reply:
In the before screenshot. The button class text is capitalized. In the after screenshot, it is not. So that's what I mean.

Reply:
I guess it is not blocking for this PR tho